server/game/cards/02_SHD/units/EmboStoicAndResolute.ts

```diff
@@ -0,0 +1,31 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { Location, WildcardCardType } from '../../../core/Constants';
+
+export default class EmboStoicAndResolute extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '0518313150',
+            internalName: 'embo#stoic-and-resolute'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'If the defender was defeated, heal up to 2 damage from a unit',
+            when: {
+                onAttackCompleted: (event, context) => event.attack.attacker === context.source,
+            },
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.Unit,
+                immediateEffect: AbilityHelper.immediateEffects.conditional({
+                    condition: (context) => context.event.attack.target.location === Location.Discard,
+                    onTrue: AbilityHelper.immediateEffects.heal({ amount: 2 }),
```

Comment:
Since the ability specifically says "up to 2" I think we instead need to do this similar to Midnight Repairs, which uses `distributeHealingAmong()`. Unfortunately that system currently doesn't have the feature to let you limit the maximum number of targets that can be chosen, so either we need to update that in this PR or push this unit to another one